packages/effection/src/context.js

```diff
@@ -113,7 +112,7 @@ Thanks!`);
     if (this.isRunning) {
       this.result = value;
     }
-    if (this.requiredChildren.size > 0) {
+    if (Array.from(this.children).some((c) => (!this.parent || c !== value) && c.isRequired)) {
```

Comment:
This conditional is quite complicated. Not sure how to reformulate this in a better way though.